File: ayesha/Basics of Programming/2D Arrays/wakanda-2.cpp
#include <iostream>
#include <vector>
using namespace std;

void upperDiagonal_traverse(int n, vector<vector<int>>& arr)
{
  // write your code here
  for(int gap=0; gap<n; gap++){
        for(int i=0, j=gap;j<n; i++,j++){
            cout<<arr[i][j]<<"\n";
        }
  }
}

int main()
{
  int n;
  cin >> n;
  vector<vector<int>> arr(n, vector<int>(n));
  for (int i = 0; i < n; i++)
    for (int j = 0; j < n; j++)
      cin >> arr[i][j];

  upperDiagonal_traverse(n, arr);
}

File: V.Vennela/Basics of Programming/Arrays2D.java/wakanda2.java
import java.io.*;
import java.util.*;

public class wakanda2{

    public static void main(String[] args) throws Exception {
        // write your code here
        Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int a[][] = new int[n][n];
    for(int i=0;i<n;i++)
    {
         for(int j=0;j<n;j++)
        {
            a[i][j]=sc.nextInt();
        }
    }
    for(int i=n-1;i>=0;i--)
    {
        int j=0,k=n-i-1;
        while(k<n && j<=i)
        {
            System.out.println(a[j][k]);
            j++;
            k++;
        }
    }
    }

}
